### PR TITLE
Remove nvm

### PR DESCRIPTION
**Pull request by rdavison**
Issue: #383 

CHANGES
- Assumes `nvm` is correctly installed within a `bash` shell.
- Added hints for users using alternative shells in CONTRIBUTING.md.

NOTE: The reason the name of the branch conflicts with the name of the PR is because this work was originally done for issue #387. During the course of development, we decided to have it address issue #383 instead. I do not know how to rename a branch within an open PR.  If anyone knows how to do it, please feel free to update the name of the brach.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -60,9 +60,7 @@ suggests the following:
 
 The site build process makes use of `nvm`. Please consult the appropriate documentation for [installing `nvm`](https://github.com/nvm-sh/nvm#installing-and-updating) for your operating system. Restart or reload your terminal to pickup the changes.
 
-#### esy
-
-The site makes use of `esy`. We currently use version 0.6.8. Please consult the `esy` documentation for installation instructions. We recommend installing `esy` globally to provide a smoother command line experience.
```
Comment (rdavison) [inline CONTRIBUTING.md:65]: Unrelated change. Noticed that these docs are no longer true.  Esy is installed during `make install-deps`. It does not need to be installed globally.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -60,9 +60,7 @@ suggests the following:
...
+NOTE: The `Makefile` assumes that `nvm` is installed and available through a `bash` shell. If you are using an alternative shell such as `zsh` or `fish`, you may need to change the `SHELL` and/or `.SHELLFLAGS` variables in order for the `Makefile` to get access to the `nvm` shell function.
```
Comment (ghost) [inline CONTRIBUTING.md]: We discussed the alternatives to `bash` on Slack. I would either:

1. Change these instructions to go all in on `bash` and request that people update `.bashrc` and ensure the test `bash -ic 'nvm --version'` works, regardless of which shell that someone uses.
OR
2. Provide instructions for overriding `SHELL` from the command line, so that users don't modify the Makefile.

**Review by ghost — APPROVED** [review]
I think this PR strikes a good compromise.

File: CONTRIBUTING.md (hunk 2 above)
Comment (rdavison) [reply CONTRIBUTING.md]: I think we should be prescriptive here and tell people to use bash, since bash is used everywhere.  If someone wants to go against the grain, then that's up to them.

Comment (rdavison) [reply CONTRIBUTING.md]: I updated the docs to reflect this: https://github.com/ocaml/v3.ocaml.org/pull/396/commits/00225b18f60fb6136124ab8fafa7efa6116bf059

File: Makefile [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+
 ifeq ($(VERCEL), 1)
   # The yarn version is picked from .engines in package.json
   YARN=yarn
   # Put .esy in node_modules for caching in Vercel
   ESY=export ESY__PREFIX=$$PWD/node_modules/.esy && npx esy
   BSB=npx bsb
 else
-  NVM=source $$NVM_DIR/nvm.sh && nvm
+  NVM=nvm
```
Comment (agarwal) [inline Makefile]: Defining `NVM` seems unnecessary now. Otherwise, we could ask why `npx` isn't also defined with a variable.

Comment (rdavison) [reply Makefile]: Good catch. Fixed: https://github.com/ocaml/v3.ocaml.org/pull/396/commits/84acf03528c24da6da7a30a01b2619d3031dc296

**Comment by agarwal** [comment]
Unfortunately, this is not working for me on my Mac M1.

```
$ bash -ic 'nvm --version'
0.35.3

$ make
nvm install
make: nvm: No such file or directory
make: *** [install-deps] Error 1
```

**Comment by rdavison** [comment]
@agarwal: Which version of Make are you using?

**Comment by agarwal** [comment]
> Which version of Make are you using?
```
$ make --version
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
```

**Comment by ghost** [comment]
For completeness, we should log the output from running `/bin/bash -ic 'nvm install'` on the problematic machine. I assume it will succeed.

**Comment by rdavison** [comment]
We had internal discussion and decided it would be for the better to remove `nvm` as the single dependency, instead relying on `npx`, which is provided when you install the node toolchain.

As of https://github.com/ocaml/v3.ocaml.org/pull/396/commits/096e7e72ae5f5b5793ba78488b390b4e8f598082 the docs have been updated to account for the changes. Following is a demonstration of the recommended workflow, as observed on my Macbook Pro (Late 2013) model running Macos 10.14 (Mojave):

- Step 0. Demonstrate that `npx` is not installed.

```bash$ make
$ make
npx yarn@1.22 install
/bin/bash: npx: command not found
make: *** [install-deps] Error 127
```

- Step 1. User installs `npx` as per the *Prerequisites* section of `CONTRIBUTING.md`
  - (optional, but recommended by `CONTRIBUTING.md`) If the user has a node version manager such as `nvm` or `fnm` installed, they can run the following to make their lives easier (note: `.nvmrc` is gitignored).
    ```bash
    $ echo 14 > .nvmrc
    $ fnm install
    Installing Node v14.17.0
    ```
- Step 2. Install, build, watch, etc...
```
$ make
npx yarn@1.22 install
yarn install v1.22.10
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
[3/5] 🚚  Fetching packages...
[4/5] 🔗  Linking dependencies...
warning " > @glennsl/bs-json@5.0.2" has incorrect peer dependency "bs-platform@^5.0.4".
warning " > autoprefixer@10.2.5" has unmet peer dependency "postcss@^8.1.0".
warning " > tailwindcss@2.1.2" has unmet peer dependency "postcss@^8.0.9".
warning "tailwindcss > postcss-nested@5.0.5" has unmet peer dependency "postcss@^8.1.13".
[5/5] 🔨  Building fresh packages...

✨  Done in 2.36s.
<...snip...>
- started server on 0.0.0.0:3000, url: http://localhost:3000
Starting up http-server, serving out
Available on:
  http://127.0.0.1:8080
  http://192.168.1.114:8080
Hit CTRL-C to stop the server
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -72,14 +83,14 @@ Run the following command to install various dependencies:
...
-The command installs the appropriate versions of node and yarn. It also installs
+The command installs the appropriate versions of node, yarn, and esy. It also vendors in the `ood` library.
 
 ### Development
 
 Use the following command to run the ReScript compiler in watch mode as well as the NextJS dev server:
 
 ```
-make watch
+make watch-and-serve
```
Comment (ghost) [inline CONTRIBUTING.md:99]: I would delete the watch-and-serve target. A while back we talked about it briefly. watch-and-serve would only be correct if we introduced some other logic to repeatedly run build and serve on each file change.

Comment (rdavison) [reply CONTRIBUTING.md:99]: Created new issue to track this: https://github.com/ocaml/v3.ocaml.org/issues/406#issue-911144595

File: Makefile [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
 else
-  NVM=source $$NVM_DIR/nvm.sh && nvm
   # Yarn version specified here because it can't bootstrap itself as a devDependency with npx.
-  YARN=$(NVM) use && npx yarn@1.22
-  ESY=$(NVM) use && npx esy
-  BSB=$(NVM) use && npx bsb
+  YARN=npx yarn@1.22
+  ESY=npx esy
+  BSB=npx bsb
 endif
```
Comment (ghost) [inline Makefile:13]: I was re-reading the "npx for yarn" issue and it mentions an interesting footnote. It appears as though we can use "yarn run esy" and "yarn run bsb". "yarn run" is less powerful than "npx". I am unsure if that is a pro or a con for this use case. I haven't tested this alternative myself. I will close this comment as resolved since there is no action needed.

**Comment by agarwal** [comment]
I tested this on my existing clone on my Mac. Also, tested a fresh clone on my Mac and a Linux machine. Everything works, so I'm going to merge this.
